File: scripts/fix-enum-literals.js
const fs = require("fs");
const path = require("path");

// File path
const filePath = path.join(__dirname, "..", "src", "data", "imported-questions-master.ts");

console.log("Starting enum literal fixes...");

// Read the file
let content = fs.readFileSync(filePath, "utf8");
console.log(`File size: ${content.length} characters`);

// Count original occurrences
const difficultyBeginner = (content.match(/"difficulty": "Beginner",/g) || []).length;
const difficultyIntermediate = (content.match(/"difficulty": "Intermediate",/g) || []).length;
const difficultyAdvanced = (content.match(/"difficulty": "Advanced",/g) || []).length;

const categoryPlatform = (content.match(/"category": "Platform Fundamentals",/g) || []).length;
const categoryConsole = (content.match(/"category": "Console Procedures",/g) || []).length;
const categoryTroubleshooting = (content.match(/"category": "Troubleshooting",/g) || []).length;
const categoryPractical = (content.match(/"category": "Practical Scenarios",/g) || []).length;
const categoryLinear = (content.match(/"category": "Linear Chain Architecture",/g) || []).length;

console.log(
  `Found difficulty strings: Beginner(${difficultyBeginner}), Intermediate(${difficultyIntermediate}), Advanced(${difficultyAdvanced})`
);
console.log(
  `Found category strings: Platform(${categoryPlatform}), Console(${categoryConsole}), Troubleshooting(${categoryTroubleshooting}), Practical(${categoryPractical}), Linear(${categoryLinear})`
);

// Replace difficulty strings
content = content.replace(/"difficulty": "Beginner",/g, '"difficulty": Difficulty.BEGINNER,');
content = content.replace(
  /"difficulty": "Intermediate",/g,
  '"difficulty": Difficulty.INTERMEDIATE,'
);
content = content.replace(/"difficulty": "Advanced",/g, '"difficulty": Difficulty.ADVANCED,');

// Replace category strings
content = content.replace(
  /"category": "Platform Fundamentals",/g,
  '"category": QuestionCategory.PLATFORM_FUNDAMENTALS,'
);
content = content.replace(
  /"category": "Console Procedures",/g,
  '"category": QuestionCategory.CONSOLE_PROCEDURES,'
);
content = content.replace(
  /"category": "Troubleshooting",/g,
  '"category": QuestionCategory.TROUBLESHOOTING,'
);
content = content.replace(
  /"category": "Practical Scenarios",/g,
  '"category": QuestionCategory.PRACTICAL_SCENARIOS,'
);
content = content.replace(
  /"category": "Linear Chain Architecture",/g,
  '"category": QuestionCategory.LINEAR_CHAIN,'
);

// Write the file back
fs.writeFileSync(filePath, content, "utf8");

console.log("Enum literal fixes completed!");

// Verify changes
const newContent = fs.readFileSync(filePath, "utf8");
const remainingDifficultyStrings = (newContent.match(/"difficulty": "[^"]*",/g) || []).length;
const remainingCategoryStrings = (newContent.match(/"category": "[^"]*",/g) || []).length;

console.log(`Remaining difficulty strings: ${remainingDifficultyStrings}`);
console.log(`Remaining category strings: ${remainingCategoryStrings}`);

if (remainingDifficultyStrings === 0 && remainingCategoryStrings === 0) {
  console.log("✅ All enum conversions completed successfully!");
} else {
  console.log("⚠️ Some string literals may still remain");
}

File: scripts/fix-enum-literals.py
#!/usr/bin/env python3
"""
Fix enum string literals in imported-questions-master.ts
This script handles the large file that couldn't be processed through Claude's Edit tools
"""

import re
import os

def fix_enum_literals():
    # File path
    script_dir = os.path.dirname(os.path.abspath(__file__))
    file_path = os.path.join(script_dir, '..', 'src', 'data', 'imported-questions-master.ts')
    
    print("Starting enum literal fixes...")
    print(f"Target file: {file_path}")
    
    # Read the file
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    print(f"File size: {len(content)} characters")
    
    # Count original occurrences
    difficulty_beginner = len(re.findall(r'"difficulty": "Beginner",', content))
    difficulty_intermediate = len(re.findall(r'"difficulty": "Intermediate",', content))
    difficulty_advanced = len(re.findall(r'"difficulty": "Advanced",', content))
    
    category_platform = len(re.findall(r'"category": "Platform Fundamentals",', content))
    category_console = len(re.findall(r'"category": "Console Procedures",', content))
    category_troubleshooting = len(re.findall(r'"category": "Troubleshooting",', content))
    category_practical = len(re.findall(r'"category": "Practical Scenarios",', content))
    category_linear = len(re.findall(r'"category": "Linear Chain Architecture",', content))
    
    print(f"Found difficulty strings: Beginner({difficulty_beginner}), Intermediate({difficulty_intermediate}), Advanced({difficulty_advanced})")
    print(f"Found category strings: Platform({category_platform}), Console({category_console}), Troubleshooting({category_troubleshooting}), Practical({category_practical}), Linear({category_linear})")
    
    # Replace difficulty strings
    content = re.sub(r'"difficulty": "Beginner",', '"difficulty": Difficulty.BEGINNER,', content)
    content = re.sub(r'"difficulty": "Intermediate",', '"difficulty": Difficulty.INTERMEDIATE,', content)
    content = re.sub(r'"difficulty": "Advanced",', '"difficulty": Difficulty.ADVANCED,', content)
    
    # Replace category strings
    content = re.sub(r'"category": "Platform Fundamentals",', '"category": QuestionCategory.PLATFORM_FUNDAMENTALS,', content)
    content = re.sub(r'"category": "Console Procedures",', '"category": QuestionCategory.CONSOLE_PROCEDURES,', content)
    content = re.sub(r'"category": "Troubleshooting",', '"category": QuestionCategory.TROUBLESHOOTING,', content)
    content = re.sub(r'"category": "Practical Scenarios",', '"category": QuestionCategory.PRACTICAL_SCENARIOS,', content)
    content = re.sub(r'"category": "Linear Chain Architecture",', '"category": QuestionCategory.LINEAR_CHAIN,', content)
    
    # Write the file back
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print("Enum literal fixes completed!")
    
    # Verify changes
    with open(file_path, 'r', encoding='utf-8') as f:
        new_content = f.read()
    
    remaining_difficulty_strings = len(re.findall(r'"difficulty": "[^"]*",', new_content))
    remaining_category_strings = len(re.findall(r'"category": "[^"]*",', new_content))
    
    print(f"Remaining difficulty strings: {remaining_difficulty_strings}")
    print(f"Remaining category strings: {remaining_category_strings}")
    
    if remaining_difficulty_strings == 0 and remaining_category_strings == 0:
        print("✅ All enum conversions completed successfully!")
        return True
    else:
        print("⚠️ Some string literals may still remain")
        return False

if __name__ == "__main__":
    success = fix_enum_literals()
    if success:
        print("\n🎯 COMPLETE: All TypeScript enum errors should now be fixed!")
    else:
        print("\n⚠️ Manual review may be needed for remaining patterns")